ospf6d/ospf6_asbr.c

```diff
@@ -2736,8 +2740,13 @@ ospf6_summary_add_aggr_route_and_blackhole(struct ospf6 *ospf6,
 					   struct ospf6_external_aggr_rt *aggr)
 {
 	struct ospf6_route *rt_aggr;
+	struct ospf6_route *old_rt = NULL;
 	struct ospf6_external_info *info;
 
+	/* Check if a route is already present. */
+	if (aggr->route)
```

Comment:
I'm curious: is there a specific reason why the resource removal was delayed? Couldn't we just release it here instead of delaying it?

Reply:
This way we can keep delay between old route removal and new route addition to a minimum I guess.